infrastructure/containers/postgres/automatic-backup/backup_or_restore.py

```diff
@@ -56,16 +60,50 @@ def run_shell_command(command, env_vars):
 ###################
 
 
-def make_backup_fpath():
-    return BACKUP_DIR / datetime.now().strftime(FILENAME_DATETIME_FORMAT)
+def make_object_name():
+    return datetime.now().strftime(FILENAME_DATETIME_FORMAT)
 
 
-def backup_database(db_backup_gzip_fpath):
+def make_backup_fpath(object_name):
+    return BACKUP_DIR / object_name
+
+
+def create_s3_client():
+    """Creates and returns an AWS S3 client for uploading backup to the cloud."""
+    sess = boto3.session.Session(
+        aws_access_key_id=BACKUP_DB__AWS_ACCESS_KEY_ID,
+        aws_secret_access_key=BACKUP_DB__AWS_SECRET_ACCESS_KEY,
+    )
+    return sess.client("s3")
+
+
+def create_s3_session():
+    """Creates and returns an AWS session for listing and downloading backups in S3."""
+    sess = boto3.session.Session(
+        aws_access_key_id=BACKUP_DB__AWS_ACCESS_KEY_ID,
+        aws_secret_access_key=BACKUP_DB__AWS_SECRET_ACCESS_KEY,
+    )
+    return sess
+
+
+def upload_backup_to_s3(s3_client, backup_fpath, backup_bucket_name, backup_object_name):
+    """Uploads the backup_fpath file to AWS S3."""
+    with open(str(backup_fpath), "rb") as f:
+        s3_client.upload_fileobj(f, backup_bucket_name, backup_object_name)
+
+
+def delete_local_backup_file(backup_fpath):
+    """Deletes the local backup file once it has been uplaoded to S3."""
+    os.remove(str(backup_fpath))
+
+
+def backup_database(object_name):
     # make sure the backup directory exists
+    db_backup_gzip_fpath = make_backup_fpath(object_name)
     db_backup_gzip_fpath.parent.mkdir(parents=True, exist_ok=True)
 
     # backup the database
-    print("Backing up database to", db_backup_gzip_fpath)
+    print("Creating the backup")
```

Comment:
Not necessary, but you could do a formatted `f""` string and output the full path to where the backup will be written to disk.